docs/painless/painless-contexts/painless-runtime-fields-context.asciidoc

```diff
@@ -0,0 +1,218 @@
+[[painless-runtime-fields-context]]
+=== Runtime fields context
+beta::[]
+Use a Painless script to return a calculated value for a
+<<painless-runtime-fields,runtime field>>, defined either in the mappings or as
+part of a query. You can also use runtime fields to override values for indexed
+fields.
+
+*Variables*
+
+`params` (`Map`, read-only)::
+        User-defined parameters passed in as part of the query.
+
+`doc` (`Map`, read-only)::
+        Contains the fields of the specified document where each field is a
+        `List` of values.
+
+{ref}/mapping-source-field.html[`params['_source']`] (`Map`, read-only)::
+        Contains extracted JSON in a `Map` and `List` structure for the fields
+        existing in a stored document.
+
+*Return*
+
+`void`::
+        No expected return value.
+
+*API*
+
+Both the standard <<painless-api-reference-shared, Painless API>> and
+<<painless-api-reference-field, Specialized Field API>> are available.
+
+
+*Example*
+
+Let's break down each of the steps from the
+<<painless-context-examples,context examples>> to understand how runtime fields
+work within the context of Painless.
+
+[[painless-runtime-fields-mappings]]
+==== Create the index mappings
```

Comment:
I'm worried that having the full example setup as part of this page is confusing since the current setup for all the other contexts requires setup from the original page with a link there. Would the longer term plan be to do this for all the context pages so there's no multi-page requirement?

Reply:
The main reason I included the examples is that getting started with the current [context examples](https://www.elastic.co/guide/en/elasticsearch/painless/7.11/painless-context-examples.html) puts more work on the user. They have to download a file, link off to other pages, and use both curl requests and Console examples to get things working. I wanted to streamline that process so that users can set up this entire example without leaving the page and navigating around.

Going forward, we can either improve the context examples page or integrate the examples directly into each context page. There are pros and cons to both approaches, and I'd like to chat with you further about which direction to take.

Reply:
My initial reaction is let's just integrate the examples into each page as that seems like it's simplest for the user and get rid of any downloaded files. I thought at the time it might be useful to have a larger data set for aggregations examples, but that's proven to not be the case.

Reply:
I have some ideas about how we can approach the examples. Let's chat about the different ways we can solve this problem and do that work in a separate PR.